sources/commands/option_external.c

```diff
@@ -28,14 +33,33 @@ static char	*find_executable(char *tgt, t_dlist *path)
 	return (0);
 }
 
-static int	run_executable(char *fullpath, t_param *param)
+static int	run_executable(char *fullpath, t_treenode *node, t_param *param)
 {
-	char	*argv[2];
+	char	**argv;
+	int		argc;
+	int		i;
+	char	**envvars;
 
+	argc = node->children_qtty;
+	argv = ft_calloc_if(sizeof(char *) * (argc + 2), 1);
```

Comment:
argv is fullpath plus those "one", "two   three", "four" from above, plus nullptr